chain-impl-mockchain/src/certificate/update_proposal.rs

```diff
@@ -0,0 +1,114 @@
+use crate::fragment::config::ConfigParams;
+use crate::transaction::SingleAccountBindingSignature;
+
+use crate::{
+    certificate::CertificateSlice,
+    key::BftLeaderId,
+    transaction::{Payload, PayloadAuthData, PayloadData, PayloadSlice},
+};
+use chain_core::{
+    mempack::{ReadBuf, ReadError, Readable},
+    property::{self, Serialize},
+};
+use typed_bytes::{ByteArray, ByteBuilder};
+
+pub type UpdateProposerId = BftLeaderId;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct UpdateProposal {
+    changes: ConfigParams,
+    proposer_id: UpdateProposerId,
+}
+
+impl UpdateProposal {
+    pub fn new(changes: ConfigParams, proposer_id: UpdateProposerId) -> Self {
+        Self {
+            changes,
+            proposer_id,
+        }
+    }
+
+    pub fn changes(&self) -> &ConfigParams {
+        &self.changes
+    }
+
+    pub fn proposer_id(&self) -> &UpdateProposerId {
+        &self.proposer_id
+    }
+
+    pub fn serialize_in(&self, bb: ByteBuilder<Self>) -> ByteBuilder<Self> {
```

Comment:
Maybe we need to move this function into the `property::Serialize` trait. But it is not related to this PR.